contracts/AirDrop.sol

```diff
@@ -0,0 +1,252 @@
+        emit DepositEther(msg.value);
+    }
+
+    /**
+     * @dev make tokens available for members to withdraw
+     *
+     * @param recepients_ - address of member
+     * @param amount_ - how many tokens will be available
+     * @param deadline_ - how long signature will be valid
+     * @param v_ - can be only 27 or 28
+     * @param r_ - contain info to recover signer
+     * @param s_ - contain info to recover signer
+     *
+     * NOTE : more information 'EIP712'
+     */
+    function dropTokens(
+        address[] calldata recepients_,
+        uint256[] calldata amount_,
+        uint256 deadline_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) external override onlyOwner {
+        require(
+            recepients_.length == amount_.length,
+            "Error : arrays have different length"
+        );
+        bytes32 structHash = keccak256(
+            abi.encode(
+                DROP_HASH,
+                keccak256(abi.encodePacked(amount_)),
+                deadline_,
+                keccak256(abi.encodePacked(recepients_)),
+                false
+            )
+        );
+        bytes32 digest = EIP712._hashTypedDataV4(structHash);
+
+        address msgSigner = ECDSA.recover(digest, v_, r_, s_);
+        require(msg.sender == msgSigner, "invalid signature");
+        require(
+            block.timestamp < deadline_,
+            "Error : signed transaction expired"
+        );
+
+        for (uint256 i; i < recepients_.length; i++) {
+            require(
+                amount_[i] != 0 && recepients_[i] != address(0),
+                "Error :'recepients_' or 'amount_' equal to 0"
+            );
+            balanceOfTokens[recepients_[i]] += amount_[i];
+        }
+        emit DropTokens(recepients_, amount_, deadline_);
+    }
+
+    /**
+     * @dev make ethers available for members to withdraw
+     *
+     * @param recepients_ - address of member
+     * @param amount_ - how many ethers will be available
+     * @param deadline_ - how long signature will be valid
+     * @param v_ - can be only 27 or 28
+     * @param r_ - contain info to recover signer
+     * @param s_ - contain info to recover signer
+     *
+     * NOTE : more information 'EIP712'
+     */
+    function dropEther(
+        address[] calldata recepients_,
+        uint256[] calldata amount_,
+        uint256 deadline_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) external override onlyOwner {
+        require(
+            recepients_.length == amount_.length,
+            "Error : arrays have different length"
+        );
+        bytes32 structHash = keccak256(
+            abi.encode(
+                DROP_HASH,
+                keccak256(abi.encodePacked(amount_)),
+                deadline_,
+                keccak256(abi.encodePacked(recepients_)),
+                true
+            )
+        );
+        bytes32 digest = EIP712._hashTypedDataV4(structHash);
```

Comment:
you imitate EIP712, so it's simple call _hashTypedDataV4(structHash); without EIP712